### DISCUSSION TITLE
Is there a way to capture request timeout errors globally instead of using try-catch for every API call?

### DISCUSSION THREAD

**Discussion by dodicandra** [sindresorhus/ky#517] [Q&A] [answered]
I have a scenario where I need to handle timeout errors globally instead of using `try-catch` or capturing errors with `react-query` in `onError`. It becomes quite cumbersome to handle these timeout errors one by one. I hope there's a solution for this, perhaps by forwarding timeout errors to `hooks` `afterResponse`.

**Comment by sholladay** [comment, marked as answer]
The easiest way is probably to create a wrapper around Ky's API.

If there were a built-in way to do this, what would you want Ky to do instead of throwing? Should it return nothing? That could make debugging difficult.

**Reply by dodicandra** [reply]
I think it still throws an error, but this timeout error is triggered within the `afterResponse` hook. Currently, the timeout error is not being passed to any other hook.
